code/ATMOSPHERICS/components/binary_devices/passive_gate.dm

```diff
@@ -155,3 +155,14 @@
 /obj/machinery/atmospherics/binary/passive_gate/power_change()
 	..()
 	update_icon()
+
+/obj/machinery/atmospherics/binary/passive_gate/npc_tamper_act(mob/living/L)
+	if(prob(50)) //Turn on/off
+		on = !on
+		investigation_log(I_ATMOS,"was turned [on ? "on" : "off"] by [L] ([key_name(L)])")
```

Comment:
Key name already shows the name, and if no key is associated it should still work.

The `by [L]` shouldn't be needed anywhere.